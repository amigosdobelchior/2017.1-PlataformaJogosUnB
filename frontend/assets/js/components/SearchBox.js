import _ from 'lodash'
import React, { Component } from 'react'
import { Search, Grid } from 'semantic-ui-react'

export default class SearchBox extends Component {
    loadGameFromServer () {
        fetch("/api/list/",
            {
                "headers": new Headers({
                    "Accept": "application/json",
                    "Content-Type": "application/json",
                }),
                "method": "GET"
            }).
          then((response) => response.json()).
          then((games) => {
              this.setState({games});
          });
    }

    constructor(props) {
        super(props);
        this.resetComponent = this.resetComponent.bind(this)
        this.handleResultSelect = this.handleResultSelect.bind(this)
        this.handleSearchChange = this.handleSearchChange.bind(this)
    }

    componentWillMount() {
        this.loadGameFromServer()
        this.resetComponent()
    }

    resetComponent() {
        this.setState({ isLoading: false, results: [], value: '' })
    }

    handleResultSelect(e, result) {
        this.setState({ value: result.name })
        window.location = `/games/${result.pk}`;
    }

    handleSearchChange(e, value) {
        this.setState({ isLoading: true, value })

        setTimeout(() => {
            if (this.state.value.length < 1) return this.resetComponent()

            const re = new RegExp(_.escapeRegExp(this.state.value), 'i')
            const isMatch = (result) => re.test(result.name)

            this.setState({
                isLoading: false,
                results: _.filter(this.state.games, isMatch),
            })
        }, 500)
    }

    render() {
        const { isLoading, value, results } = this.state

        return (
            <Grid>
                <Grid.Column width={8}>
                    <Search
                        size='small'
                        placeholder='Search...'
                        loading={isLoading}
                        onResultSelect={this.handleResultSelect}
                        onSearchChange={this.handleSearchChange}
                        results={results}
                        value={value}
                        {...this.props}
                    />
                </Grid.Column>
            </Grid>
        )
    }
}