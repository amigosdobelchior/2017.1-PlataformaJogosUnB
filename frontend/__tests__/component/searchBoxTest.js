import React from 'react';
import SearchBox from '../../assets/js/components/SearchBox';
import {Search, Grid} from "semantic-ui-react";
import renderer from 'react-test-renderer';
var expect = required('expect');

test('Test render SearchBox', () => {
  const component = renderer.create(
      <Grid>
          <Grid.Column width={8}>
              <Search
                  size='small'
                  placeholder='Pesquisar...'
                  loading='false'

                  results={['game1','game2','game3','game4','game5']}
                  value= ''

              />
          </Grid.Column>
      </Grid>
     );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
