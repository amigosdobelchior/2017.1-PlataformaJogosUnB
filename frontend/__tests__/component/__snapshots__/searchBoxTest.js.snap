// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render SearchBox 1`] = `
<div
  className="ui grid"
>
  <div
    className="eight wide column"
  >
    <div
      className="ui small loading search"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
    >
      <div
        className="ui icon input"
      >
        <input
          autoComplete="off"
          className="prompt"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          placeholder="Pesquisar..."
          tabIndex="0"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          className="search icon"
        />
      </div>
      <div
        className="results transition"
      >
        <div
          0="g"
          1="a"
          2="m"
          3="e"
          4="1"
          className="result"
          onClick={[Function]}
        >
          <div
            className="content"
          />
        </div>
        <div
          0="g"
          1="a"
          2="m"
          3="e"
          4="2"
          className="result"
          onClick={[Function]}
        >
          <div
            className="content"
          />
        </div>
        <div
          0="g"
          1="a"
          2="m"
          3="e"
          4="3"
          className="result"
          onClick={[Function]}
        >
          <div
            className="content"
          />
        </div>
        <div
          0="g"
          1="a"
          2="m"
          3="e"
          4="4"
          className="result"
          onClick={[Function]}
        >
          <div
            className="content"
          />
        </div>
        <div
          0="g"
          1="a"
          2="m"
          3="e"
          4="5"
          className="result"
          onClick={[Function]}
        >
          <div
            className="content"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
